chainer/variable.py

```diff
@@ -515,6 +515,15 @@ def debug_print(self):
         with cuda.get_device_from_array(self.data) as dev:
             xp = numpy if int(dev) == -1 else cuda.cupy
 
+            if self.data is None:
+                backend = shape = dtype = stats = None
```

Comment:
* Does it make sense to print `device: CPU` when data is None?
* Please add a test case for this.

Reply:
- Changed to `device: None`.
- Added test.